compiler/tools/suggest.nim

```diff
@@ -477,6 +481,115 @@ proc findTrackedSym*(g: ModuleGraph;): PSym =
     # xxx: the node finding should be specialized per symbol kind
     result = findTrackedNode(m.ast, g.config.m.trackPos)
 
+proc getSymNode(node: ParsedNode): ParsedNode =
+  result = node
+  if result.kind == pnkPostfix:
+    result = result[^1]
+  elif result.kind == pnkPragmaExpr:
+    result = getSymNode(result[0])
+
+proc pnkToSymKind(kind: ParsedNodeKind): TSymKind =
+  result = skUnknown
+  case kind
+  of pnkConstSection, pnkConstDef: result = skConst
+  of pnkLetSection: result = skLet
+  of pnkVarSection: result = skVar
+  of pnkProcDef: result = skProc
+  of pnkFuncDef: result = skFunc
+  of pnkMethodDef: result = skMethod
+  of pnkConverterDef: result = skConverter
+  of pnkIteratorDef: result = skIterator
+  of pnkMacroDef: result = skMacro
+  of pnkTemplateDef: result = skTemplate
+  of pnkTypeDef, pnkTypeSection: result = skType
+  else: discard
+
+proc getName(node: ParsedNode): string =
+  if node.kind == pnkIdent:
+    result = node.startToken.ident.s
+  elif node.kind == pnkAccQuoted:
+    result = "`"
+    for t in node.idents:
+      result.add t.ident.s
+    result.add "`"
+
+proc processFlags(sug: Suggest; n: ParsedNode) =
+  var
+    identDeprecated: bool
+    colonDeprecated: bool
+  for s in n.sons:
+    identDeprecated = s.kind == pnkIdent and getName(s) == "deprecated"
+    colonDeprecated = s.kind == pnkExprColonExpr and getName(s[0]) == "deprecated" 
+    if identDeprecated or colonDeprecated:
+      sug.flags.incl SuggestFlag.deprecated
+
+proc parsedNodeToSugget(n: ParsedNode; originKind: ParsedNodeKind; module: string): Suggest =
+  if n.kind in {pnkError, pnkEmpty}: return
+  if n.kind notin {pnkProcDef..pnkVarTuple}: return
+  new(result)
+  var token = getToken(n)
+  var name = ""
+
+  if n.kind in pnkRoutineDefs and n[pragmasPos].kind == pnkPragma:
+    processFlags(result, n[pragmasPos])
+  elif n[0].kind == pnkPragmaExpr and n[0][^1].kind == pnkPragma:
+    processFlags(result, n[0][^1])
+
+  if n.kind != pnkVarTuple:
+    var node: ParsedNode = getSymNode(n[0])
+    token = getToken(node)
```

Comment:
```suggestion
    let node = getToken(getSymNode(n[0]))
```
Immutability is generally preferred over repeated assignments, as it makes the code easier to comprehend.